main.js

```diff
@@ -0,0 +1,180 @@
+    img.setAttribute("alt", "rain");
+  } else if (chanceofsnow > 50) {
+    // img = '<img src="./assets/icons8-light-snow.gif" alt="snow" />';
+    img.setAttribute("src", "./assets/icons8-light-snow.gif");
+    img.setAttribute("alt", "snow");
+  }
+  mainArticle.append(img);
+
+  const mainAside = document.querySelector("#main-aside")
+
+  // This will empty the existing element contents so that we can recreate them.
+  // This is an alternative to using replaceWith(). Not sure if I like it...
+  mainAside.innerHTML = '';
+
+  // Add 3 article elements under main aside to get today, tomorrow, etc data
+  const article1 = document.createElement("article");
+  const article2 = document.createElement("article");
+  const article3 = document.createElement("article");
+  function addDaysInfo(parent, article, dayName, dayNum) {
+    const h3 = document.createElement("h3");
+    h3.innerText = dayName;
+    const p5 = document.createElement("p");
+    const p6 = document.createElement("p");
+    const p7 = document.createElement("p");
+
+    const avgTemp = weatherInfo.weather[dayNum].avgtempF;
+    p5.innerHTML =
+      "<strong>Average Temperature: </strong>" + avgTemp + " °F";
+    const maxTemp = weatherInfo.weather[dayNum].maxtempF;
+    p6.innerHTML =
+      "<strong>Maximum Temperature: </strong>" + maxTemp + " °F";
+    const minTemp = weatherInfo.weather[dayNum].mintempF;
+    p7.innerHTML =
+      "<strong>Minimum Temperature: </strong>" + minTemp + " °F";
+
+    article.append(h3, p5, p6, p7);
+    // append article to main aside
+
+    parent.append(article);
+  }
+
+  addDaysInfo(mainAside, article1, "Today", 0);
+  addDaysInfo(mainAside, article2, "Tomorrow", 1);
+  addDaysInfo(mainAside, article3, "Day After Tomorrow", 2);
+
+  addPreviousSearch(city, feelsLikeF);
+}
+
+function addPreviousSearch(city, feelsLikeF) {
+  // Here we are using a class. Before we were using the attribute id.
+  // Which is the better way to do things? perhaps a question for your instructor.
+  const previousSearchList = document.querySelector(".previous-searches");
+
+  // Check for pre-existing searches with the same city name
+  // If found, return early
+  for (var i = 0; i < previousSearchList.children.length; i++) {
+    if (city == previousSearchList.children[i].getAttribute("city")) {
+      return;
+    }
+  }
+
+  const newLi = document.createElement("li");
+  previousSearchList.append(newLi);
+  // Creating the text for previous searches "X degrees fahrenheit"
+  const a = document.createElement("a");
+  a.setAttribute("href", base_url + city + format);
+  a.setAttribute("city", city);
+  newLi.setAttribute("city", city);
+  a.innerText = city;
+  newLi.append(a);
+  newLi.innerHTML += ` - ${feelsLikeF} °F`;
+  previousSearchList.append(newLi);
+
+  //This removes the "no previous searches" text after you input a location
+  document.querySelector(".hide").setAttribute("style", "display: none");
+
+  form.reset();
```

Comment:
It seems like resetting your form is the last thing you are doing but you should do it almost immediately after you grab the event value. (Id recommend in your fetch after calling createWeatherForecast())